client/src/js/components/courseEvals/CourseEvals.js

```diff
@@ -0,0 +1,300 @@
+
+const getCourseKey = course => course["_id"].substring(0, 10);
+
+async function getAllEvals (courseCode) {
+  try {
+    const res = await axios.get(wasetimeApiStatic.courseEvalsBaseURL + courseCode);
+    return res.data;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+const getRelatedCourses = (coursesList, courseCode, thisCourseKey) => {
+  // Todo: Get related courses with course code
+  // let relatedCourses = coursesList.filter(c => c.course_code === courseCode && getCourseKey(c) !== thisCourseKey);
+
+  return [
+    {
+      "_id":"2603033019012020260303301926",
+      "course_code": 'INFP31ZL',
+      "year":2020,
+      "term":"fallSem",
+      "title":"Software Engineering B",
+      "title_jp":"ソフトウェア工学B",
+      "instructor":"HAGIMOTO, Junzo/HIRANABE, Kenji/FUKAZAWA, Yoshiaki/WASHIZAKI, Hironori",
+      "instructor_jp":"萩本 順三/平鍋 健児/深澤 良彰/鷲崎 弘宜",
+      "occurrences":[
+        {
+           "day":3,
+           "start_period":5,
+           "end_period":5,
+           "building":"-1",
+           "classroom":"undecided",
+           "location":"-1-undecided"
+        }
+      ],
+      "lang":"JP",
+      "keys":[
+        {
+           "school":"FSE",
+           "key":"2603033019012020260303301926"
+        }
+      ]
+    },
+    {
+      "_id":"26GC02300301202026MC03300926",
+      "course_code": 'INFP31ZL',
+      "year":2020,
+      "term":"fallSem",
+      "title":"Software Engineering",
+      "title_jp":"Software Engineering",
+      "instructor":"ARCAINI, Paolo/KOBAYASHI, Tsutomu/FUKAZAWA, Yoshiaki/HONIDEN, Shinichi/WASHIZAKI, Hironori",
+      "instructor_jp":"アルカイニ パオロ/小林 努/深澤 良彰/本位田 真一/鷲崎 弘宜",
+      "occurrences":[
+        {
+           "day":1,
+           "start_period":3,
+           "end_period":3,
+           "building":"-1",
+           "classroom":"undecided",
+           "location":"-1-undecided"
+        }
+      ],
+      "keywords":[
+        "IPSE"
+      ],
+      "lang":"EN",
+      "keys":[
+        {
+           "school":"FSE",
+           "key":"26GC02300301202026MC03300926"
+        }
+      ]
+    }
+  ];
+}
+
+const getCourseEvals = (evaluations, courseKey) => evaluations.filter(e => e.course_key === courseKey);
+
+class CourseEvals extends React.Component {
+  state = {
+    thisCourse: {},
+    relatedCourses: [],
+    evaluations: [],
+    thisCourseEvals: [],
+    satisfaction: 0,
+    difficulty: 0,
+    benefit: 0,
+    isLoaded: false
+  };
+
+  async componentDidMount () {
+    const courseID = qs.parse(this.props.location.search, { ignoreQueryPrefix: true }).courseID;
+    const coursesList = await getAllCourses();
+    const thisCourse = getCourse(coursesList, courseID);
+    const thisCourseCode = getCourseCode(thisCourse);
+    const thisCourseKey = getCourseKey(thisCourse);
+    const relatedCourses = getRelatedCourses(coursesList, thisCourseCode, thisCourseKey);
+    const evaluations = await getAllEvals(thisCourseCode);
```

Comment:
Better naming:
`const courseEvals = await getCourseEvals(thisCourseCode);`
`getAllEvals` is a bit misleading since we're not getting all evaluations from db.

Reply:
Here is my change:
```
const evalsWithSameCode = await getCourseEvals(thisCourseCode);
const thisCourseEvals = filterCourseEvalsByKey(evalsWithSameCode, thisCourseKey);
...
this.setState({
  ...
  courseEvals: evalsWithSameCode,
  ...
})
```